packages/edit-post/src/components/layout/index.js

```diff
@@ -114,21 +118,24 @@ function useEditorStyles() {
 			} );
 		}
 
-		const baseStyles = hasThemeStyles
+		let baseStyles = hasThemeStyles
 			? editorSettings.styles ?? []
 			: defaultEditorStyles;
 
-		// Add a constant padding for the typewritter effect. When typing at the
+		// Add a constant padding for the typewriter effect. When typing at the
 		// bottom, there needs to be room to scroll up.
 		if (
 			! isZoomedOutView &&
 			! hasMetaBoxes &&
 			renderingMode === 'post-only' &&
-			! DESIGN_POST_TYPES.includes( postType )
+			! DESIGN_POST_TYPES.includes( postType ) &&
+			! baseStyles.includes( TYPEWRITER_STYLE )
 		) {
-			baseStyles.push( {
-				css: 'body{padding-bottom: 40vh}',
-			} );
+			baseStyles.push( TYPEWRITER_STYLE );
+		} else if ( baseStyles.includes( TYPEWRITER_STYLE ) ) {
+			baseStyles = baseStyles.filter(
+				( style ) => style !== TYPEWRITER_STYLE
+			);
```

Comment:
So first, I think this fix is fine and we can ship it as is if you like. 

But I have a couple thoughts:

Removing the style shouldn't be needed. I think the issue is that we're providing the value of the setting by reading the exact same settings. I think this hook shouldn't be using the "editorSettings" from the editor package. Instead it should be using the "settings" prop of the Layout component.  

The second thing is that I think that this code should probably move within the editor package somewhere. Because I believe this padding should probably be applied to the editor every time "posts/pages" are rendered in post-only mode regardless of whether it's the post editor or site editor or some other page.

---
Both of these comments can be addressed separately though.

Reply:
Thanks for the speedy review.

I can follow up on that, though it'll be next week as it's end of day for me now and I'll only be back working on Monday.

Reply:
Right, why does the `TYPEWRITER_STYLE` need to be removed if this is the only place where it's adding it to the `baseStyles`?

Reply:
It looks like it's due to the code mutating `editorSettings.styles`, so it should be an easy fix to prevent that. I've pushed some commits to the PR.

Reply:
I've gone ahead and merged the PR, but happy to revisit in a follow-up if there's any further feedback.